submissions/vovan-zt/app_friends/script.js

```diff
@@ -0,0 +1,175 @@
+﻿const url = 'https://randomuser.me/api/?results=20'
+
+const myForm = document.querySelector('.form'),
+    cards = document.querySelector('.cards'),
+    resetBtn = document.querySelector('.filter__reset-button'),
+    hamburgerBtn = document.querySelector('.hamburger'),
+    filter = document.querySelector('.filter'),
+    closeBtn = document.querySelector('.filter__close')
+
+let dataBase = []
+let sortedDataBase = []
+
+async function getResponse() {
+    try {
+        const response = await fetch(url)
+        const data = await response.json()
+        const { results } = await data
+        dataBase = [...results]
+        sortedDataBase = [...dataBase]
+        showCards(dataBase)
+    } catch (e) {
+        console.error(e)
+    }
+}
+
+function createCards({ picture, name, dob, gender, location, email, cell }) {
+    const cardSection = document.createElement('article')
+    cardSection.classList.add('cards__wrapper')
+
+    let colorGender = ''
+    let bgcolorName = ''
+    if (gender == 'female') {
+        colorGender = 'color_violet'
+        bgcolorName = 'bgcolor_violet'
+    }
+
+    cardSection.innerHTML = `
+    <div class="cards__wrapper-name ${bgcolorName}">
+    <h3> ${name.first} ${name.last}</h3>
+    </div>
+
+    <div class="cards__wrapper-photo">
+      <img src=${picture.large} alt="">
+    </div>
+
+    <div class="cards__wrapper-years">
+    I have ${dob.age} years old
+    </div>
+
+    <div class="cards__wrapper-email">
+    <a href='${email}'>${email}</a>
+    </div>
+
+    <div class="cards__wrapper-phone">
+    <a  href="tel:+ ${cell}"> ${cell}</a>
+    </div>
+
+    <div class="cards__wrapper-city">
+      ${location.city}
+    </div>
+
+    <hr class="devider">
+
+    <div class="cards__wrapper-gender ${colorGender}">${gender}</div>
+
+  `
+    cards.append(cardSection)
+}
+
+function showCards(dataBase) {
+    dataBase.forEach(
+        ({ picture, name, dob, gender, location, email, cell }) => {
+            return createCards({
+                picture,
+                name,
+                dob,
+                gender,
+                location,
+                email,
+                cell,
+            })
+        }
+    )
+}
+
```

Comment:
Avoid adding new elements on each loop iteration, better create an array of elements, and then in one `append` operation add them all to DOM.

Reply:
I initially did this, but in the process of work, difficulties arose that could not be solved. Everything is fine now, I'll fix it soon.